tests/integration/cartography/intel/gcp/test_permission_relationships.py

```diff
@@ -0,0 +1,229 @@
+from tests.data.gcp.policy_bindings import MOCK_IAM_SERVICE_ACCOUNTS
+from tests.data.gcp.policy_bindings import MOCK_POLICY_BINDINGS_RESPONSE
+from tests.integration.util import check_nodes
+from tests.integration.util import check_rels
+
+TEST_PROJECT_ID = "project-123"
+TEST_UPDATE_TAG = 123456789
+COMMON_JOB_PARAMS = {
+    "UPDATE_TAG": TEST_UPDATE_TAG,
+    "ORG_RESOURCE_NAME": "organizations/1337",
+    "PROJECT_ID": TEST_PROJECT_ID,
+    "gcp_permission_relationships_file": "dummy_path",  # Will be mocked!
+}
+GSUITE_COMMON_PARAMS = {
+    **COMMON_JOB_PARAMS,
+    "CUSTOMER_ID": "customer-123",
+}
+
+
+def _create_test_project(neo4j_session):
+    """Create a test GCP project node."""
+    neo4j_session.run(
+        """
+        MERGE (project:GCPProject{id: $project_id})
+        ON CREATE SET project.firstseen = timestamp()
+        SET project.lastupdated = $update_tag
+        """,
+        project_id=TEST_PROJECT_ID,
+        update_tag=TEST_UPDATE_TAG,
+    )
+
+
+@patch.object(
+    cartography.intel.gcp.permission_relationships,
+    "parse_permission_relationships_file",
+    return_value=MOCK_PERMISSION_RELATIONSHIPS_YAML,
+)
+@patch.object(
+    cartography.intel.gcp.compute,
+    "get_gcp_instance_responses",
+    return_value=[MOCK_COMPUTE_INSTANCES],
+)
+@patch.object(
+    cartography.intel.gcp.compute,
+    "get_zones_in_project",
+    return_value=[{"name": "us-east1-b"}],
+)
+@patch.object(
+    cartography.intel.gcp.storage,
+    "get_gcp_buckets",
+    return_value=MOCK_STORAGE_BUCKETS,
+)
+@patch.object(
+    cartography.intel.gcp.policy_bindings,
+    "get_policy_bindings",
+    return_value=MOCK_POLICY_BINDINGS_RESPONSE,
+)
+@patch.object(
+    cartography.intel.gsuite.groups,
+    "get_members_for_groups",
+    return_value=MOCK_GSUITE_GROUP_MEMBERS,
+)
+@patch.object(
+    cartography.intel.gsuite.groups,
+    "get_all_groups",
+    return_value=MOCK_GSUITE_GROUPS,
+)
+@patch.object(
+    cartography.intel.gsuite.users,
+    "get_all_users",
+    return_value=MOCK_GSUITE_USERS,
+)
+@patch.object(
+    cartography.intel.gcp.iam,
+    "get_gcp_roles",
+    return_value=MOCK_IAM_ROLES,
+)
+@patch.object(
+    cartography.intel.gcp.iam,
+    "get_gcp_service_accounts",
+    return_value=MOCK_IAM_SERVICE_ACCOUNTS,
+)
+def test_sync_gcp_permission_relationships(
+    mock_get_service_accounts,
+    mock_get_roles,
+    mock_get_all_users,
+    mock_get_all_groups,
+    mock_get_group_members,
+    mock_get_policy_bindings,
+    mock_get_buckets,
+    mock_get_zones,
+    mock_get_instance_responses,
+    mock_parse_yaml,
+    neo4j_session,
+):
+    """
+    Test that GCP permission relationships sync creates the expected nodes and relationships.
+    """
+    # ARRANGE
+    _create_test_project(neo4j_session)
+    mock_iam_client = MagicMock()
+    mock_admin_resource = MagicMock()
+    mock_storage_client = MagicMock()
+    mock_compute_client = MagicMock()
+    mock_asset_client = MagicMock()
+
+    # ACT
```

Comment:
Nit: technically this is the arrange phase since the real func under test is the permission rels